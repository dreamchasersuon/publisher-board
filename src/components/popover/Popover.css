.Popover {
  position: absolute;
  display: flex;
  flex-direction: column;
  right: 60px;
  margin-top: 50px;
  border-radius: 4px;
  padding: 5px;
  box-shadow: 0 0 20px rgba(0, 0, 0, 0.25);
  background-color: white;
}

.Popover-button {
  padding: 10px 20px;
  font-size: 14px;
  color: #444;
  text-align: left;
  border: none;
  outline: none;
  border-bottom: 1px solid #e5e5e5;
  cursor: pointer;
}

.Popover-button:last-child {
  border: none;
}
