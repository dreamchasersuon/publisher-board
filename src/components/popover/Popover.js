import React from 'react';
import PropTypes from 'prop-types';
import Button from '../buttons/button-open-modal/Button';
import './Popover.css';

export default function Popover({
  handleOnBlurPopover,
  popoverRef,
  handleOpenModal,
}) {
  return (
    <div
      data-testid="popover"
      tabIndex="0"
      ref={popoverRef}
      onBlur={handleOnBlurPopover}
      className="Popover"
    >
      <Button
        testId="update-user"
        title="Update User"
        dataName="updateUser"
        onPress={handleOpenModal}
        className="Popover-button"
      />
      <Button
        testId="get-transactions"
        title="Get Transactions"
        dataName="getTransactions"
        onPress={handleOpenModal}
        className="Popover-button"
      />
      <Button
        testId="update-balance"
        title="Update Balance"
        dataName="updateBalance"
        onPress={handleOpenModal}
        className="Popover-button"
      />
    </div>
  );
}

Popover.propTypes = {
  handleOnBlurPopover: PropTypes.func,
  handleOpenModal: PropTypes.func,
  popoverRef: PropTypes.object,
};
