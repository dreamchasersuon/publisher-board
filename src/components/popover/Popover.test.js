import React from 'react';
import snapshotDiff from 'snapshot-diff';
import Popover from './Popover';

const handleOpenModal = jest.fn();
const handleOnBlur = jest.fn();

test('renders popover or nothing if not visible', () => {
  expect(
    snapshotDiff(
      <Popover
        isVisible
        handleOpenModal={handleOpenModal}
        handleOnBlurPopover={handleOnBlur}
        popoverRef={null}
      />,
      <Popover
        isVisible={false}
        handleOpenModal={handleOpenModal}
        handleOnBlurPopover={handleOnBlur}
        popoverRef={null}
      />,
    ),
  ).toMatchSnapshot();
});
