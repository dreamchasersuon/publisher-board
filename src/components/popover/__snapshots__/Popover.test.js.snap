// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders popover or nothing if not visible 1`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;
